### PR TITLE
Fix Issue #1245: GitHub annotations does not distinguish between warnings and errors

### PR DESCRIPTION
**Pull request by eramod**
What does this PR do?
Addresses #1245   
- Makes the GitHub Actions printer distinguish between errors and warnings
- Makes the GitHub Actions printer respect the quiet flag
- Adds tests for each case

### CODE REVIEW

File: lib/printers/github-actions.js [hunk 1]
```diff
@@ -12,22 +13,26 @@ class GitHubActionsPrinter {
...
       }
     }
   }
 
-  _formatAnnotation(message, options = {}) {
+  _formatAnnotation(message, errorSeverity, options = {}) {
     message = message || '';
 
-    let output = '::error';
+    let output = errorSeverity === 2 ? '::error' : '::warning';
```
Comment (chancancode) [inline lib/printers/github-actions.js:35]:
```suggestion
    let output = errorSeverity === Linter.ERROR_SEVERITY ? '::error' : '::warning';
```

File: test/unit/github-actions-test.js [hunk 2]
```diff
@@ -10,12 +10,17 @@ describe('GitHub Actions Annotations', function () {
     });
 
     test('formats error annotations', () => {
-      let output = printer._formatAnnotation('foo', { file: 'bar.hbs', line: 4, col: 5 });
+      let output = printer._formatAnnotation('foo', 2, { file: 'bar.hbs', line: 4, col: 5 });
```
Comment (chancancode) [inline test/unit/github-actions-test.js:13]:
```suggestion
      let output = printer._formatAnnotation('foo', Linter.ERROR_SEVERITY, { file: 'bar.hbs', line: 4, col: 5 });
```

File: test/unit/github-actions-test.js [hunk 3]
```diff
@@ -10,12 +11,25 @@ describe('GitHub Actions Annotations', function () {
...
+        file: 'bar.hbs',
+        line: 4,
+        col: 5,
+      });
+      expect(output).toEqual('::warning file=bar.hbs,line=4,col=5::foo');
+    });
+
     test('escapes correctly', () => {
-      let output = printer._formatAnnotation('wow!\r\nthis is a multiline\nmessage!!', {
+      let output = printer._formatAnnotation('wow!\r\nthis is a multiline\nmessage!!', 2, {
```
Comment (chancancode) [inline test/unit/github-actions-test.js:32]:
```suggestion
      let output = printer._formatAnnotation('wow!\r\nthis is a multiline\nmessage!!', Linter.ERROR_SEVERITY, {
```

one more here

**Comment by eramod** [comment]
> Thanks for working on this! I think once @chancancode's last comment is addressed we can land this.

@rwjblue Just wanted to let you know that I addressed Godfrey's comments in my "Respond to PR feedback" commit. :)

**Comment by chancancode** [comment]
Looks good to me!

**Comment by rwjblue** [comment]
Thank you!! Released in v2.5.0!

**Pull request merged**